expr/literals.go

```diff
@@ -45,25 +45,6 @@ type (
 	Null types.Type
 )
 
-func StructLiteralFromProto(s *proto.Expression_Literal_Struct) StructLiteralValue {
```

Comment:
Since this is currently exported, removing it is a breaking change. 

My preference would be to just add a `// Deprecated: use VirtualTableExprFromProto instead` and not remove it. 

If we're going to remove this, then we need to mark this as breaking.

Reply:
ok, my thought was, though this API is public but only usage of this API can be to construct a plan from proto. So I felt this won't be used from outside substrait-go.  So I felt safe to remove this API and change usage of this to by introducing `VirtualTableExprFromLiteralProto(s *proto.Expression_Literal_Struct) VirtualTableExpressionValue`.
To signal this intent I have migrated new API introduced in this PR to "plan/internal" package in latest commit

Also,  I have accommodated changes to mark this API deprecated

Reply:
Looks good to me. Thanks for adding the `Deprecated` mark